src/lib/components/payload-input-with-encoding.svelte

```diff
@@ -0,0 +1,61 @@
+<script context="module" lang="ts">
+  const encoding = ['json/plain', 'json/protobuf'] as const;
+  export type PayloadInputEncoding = (typeof encoding)[number];
+  export const isPayloadInputEncodingType = (
+    x: any, // eslint-disable-line @typescript-eslint/no-explicit-any
+  ): x is PayloadInputEncoding => encoding.includes(x);
+</script>
+
+<script lang="ts">
+  import { type Writable } from 'svelte/store';
+
+  import { onDestroy } from 'svelte';
+
+  import Card from '$lib/holocene/card.svelte';
+  import RadioGroup from '$lib/holocene/radio-input/radio-group.svelte';
+  import RadioInput from '$lib/holocene/radio-input/radio-input.svelte';
+  import { translate } from '$lib/i18n/translate';
+
+  import PayloadInput from './payload-input.svelte';
+
+  export let input: string;
+  export let encoding: Writable<PayloadInputEncoding>;
+  export let error = false;
+  export let resetValues = false;
+  export let loading = false;
+
+  $: {
```

Comment:
This kind of feels a bit weird. It feels a bit off that we're using a state change to trigger an action. Like, I assume `resetValues` is being triggered from outside this component and that why when we're looking at the reactive statement, we're like if it's now `true` then do this action. But, I wonder if that means that the value should be stored outside of this component and be passed in?

Reply:
I think this was added so that if this is in a `Modal` the value can be reset when it's closed (cc @Alex-Tideman to confirm). I'll look into it a bit more though and see if there's a better solution!

Reply:
**Update**: I think[ this](https://github.com/temporalio/ui/pull/2407/commits/b276e06b04d556b5dc9c7c06916fea55aaa9f441) will work instead of having to trigger `resetView` outside of the `CodeBlock` component.